src/main/java/org/logstash/tcp/InputLoop.java

```diff
@@ -119,17 +135,30 @@ private static final class InputHandler extends ChannelInitializer<SocketChannel
          */
         private final SslContext sslContext;
 
+        /**
+         * Namespaced Metric.
+         */
+        private final NamespacedMetric metric;
+
         /**
          * Ctor.
          * @param decoder {@link Decoder} provided by JRuby.
          */
-        InputHandler(final Decoder decoder, final SslContext sslContext) {
+        InputHandler(final Decoder decoder, final SslContext sslContext, final NamespacedMetric metric) {
             this.decoder = decoder;
             this.sslContext = sslContext;
+            this.metric = metric;
         }
 
         @Override
         protected void initChannel(final SocketChannel channel) throws Exception {
+            Integer currentActiveConnections = activeConnections.incrementAndGet();
+            if (currentActiveConnections > peakConnections.get()) {
```

Comment:
Since no locking, `peakConnections` can be a regular int.
I think the no locking approch is better, indeed, we will get less than perfect metrics, but we will not degrade performance.